src/math.rs

```diff
@@ -259,19 +289,43 @@ impl<S: BaseFloat> Obb<S> {
     }
 }
 
+impl<S> PointCulling<S> for Obb<S>
+where
+    S: 'static + BaseFloat + Sync + Send,
+{
+    fn intersects(&self, aabb: &Aabb3<S>) -> bool {
+        intersects(&self.corners, &self.separating_axes, aabb)
+    }
+
+    fn contains(&self, p: &Point3<S>) -> bool {
+        let Point3 { x, y, z } =
+            self.isometry_inv.rotation.rotate_point(*p) + self.isometry_inv.translation;
+        x.abs() <= self.half_extent.x
+            && y.abs() <= self.half_extent.y
+            && z.abs() <= self.half_extent.z
+    }
+
+    fn transform(&self, isometry: &Isometry3<S>) -> Box<PointCulling<S>> {
+        Box::new(Self::new(
+            isometry * &self.isometry_inv.inverse(),
+            self.half_extent,
+        ))
+    }
+}
+
 #[derive(Debug, Clone)]
-pub struct OrientedBeam {
+pub struct OrientedBeam<S> {
```

Comment:
How about making this its own small PR?

Reply:
I had to change the <f64> in <S> to allow for Sync and Send fot the trait. The other changes except for the trait are  again already in master, here unfortunately yet not visible cause the PR of NodeIdIterator is still open